src/util/simplify_expr.cpp

```diff
@@ -1409,7 +1409,9 @@ bool simplify_exprt::simplify_object(exprt &expr)
   {
     const typet &op_type=ns.follow(expr.op0().type());
 
-    assert(expr.operands().size()==1);
+    DATA_INVARIANT(
+      expr.operands().size() == 1,
+      "typecasts must have exactly one argument");
```

Comment:
Introduce `const typecast_exprt &tc = to_typecast_expr(expr);` above and change the code to use `tc` with its proper API. Let's not duplicate consistency checks on expressions.

Reply:
Done.